### ISSUE TITLE
[feat]: disable ssh password authentication

### ISSUE THREAD

**Issue by AndySchroder** [Start9Labs/start-os#2319]
### Prerequisites

- [X] I have searched for [existing issues](https://github.com/start9labs/start-os/issues) that already suggest this feature.

### Describe the Feature Request

The ssh server seems to have password authentication turned on. I think this should be disabled by default. You already have ssh pubkey submission in the GUI and the start9 user has it's password disabled. I think sshd should be further hardened by not even allowing clients to attempt password authentication. Additionally, by even allowing it, it creates confusion for users who may think there is a password somewhere that can be used. If you have password authentication disabled, people will know they need to figure out how to do pubkey authentication only.

### Describe the Use Case

hardening the system

**Comment by MattDHill** [comment]
You must have a dev build. SSH password auth should be disabled in production builds.

**Comment by AndySchroder** [comment]
Using this one: https://github.com/Start9Labs/start-os/releases/tag/v0.3.4.3 .

**Comment by dr-bonez** [comment]
You are correct that ssh is configured with `PasswordAuthentication = yes` however with the only user account being locked (there is provably no valid password for the account) there is no security concern here. I sympathize with the point about confusion here, however we do not encourage ssh as a primary method of interacting with the device. If it is harder to stumble your way into the ssh prompt without reading the documentation, perhaps that is a good thing?

**Comment by AndySchroder** [comment]
Okay, I guess it's your call then how you want to design the system. I do prefer `PasswordAuthentication = no` so that you can't mess anything up and have clarity on what is going on without knowing the contents of the `shadow` file.

>  If it is harder to stumble your way into the ssh prompt without reading the documentation, perhaps that is a good thing?

Don't totally agree with this point since there is not a ton of clarity in the documentation about the validity of the password and ssh.

**Comment by kn0wmad** [comment]
> Okay, I guess it's your call then how you want to design the system. I do prefer `PasswordAuthentication = no` so that you can't mess anything up and have clarity on what is going on without knowing the contents of the `shadow` file.
> 
> > If it is harder to stumble your way into the ssh prompt without reading the documentation, perhaps that is a good thing?
> 
> Don't totally agree with this point since there is not a ton of clarity in the documentation about the validity of the password and ssh.

This [docs PR](https://github.com/Start9Labs/documentation/pull/421) will add an introduction with some clarity around SSH access.  This ticket can probably be closed.
